src/baldr/openlr.cc

```diff
@@ -1,97 +0,0 @@
-#include <cmath>
-#include <string>
-#include <vector>
-
-#include "baldr/graphconstants.h"
-#include "baldr/openlr.h"
-#include "midgard/encoded.h"
-#include "midgard/pointll.h"
-
-#include "proto/trip.pb.h"
-#include "proto/tripcommon.pb.h"
-
-namespace valhalla {
-namespace midgard {
-
-using FormOfWay = OpenLR::LocationReferencePoint::FormOfWay;
-
-FormOfWay road_class_to_fow(const TripLeg::Node& node) {
-  const bool is_missing_rc = !node.has_edge() || !node.edge().has_road_class();
-  if (is_missing_rc || !node.edge().has_use()) {
-    return FormOfWay::OTHER;
-  }
-  const TripLeg::Use use = node.edge().use();
-  const RoadClass rc = node.edge().road_class();
-  const bool one_way =
-      node.edge().has_traversability() && node.edge().traversability() != TripLeg::kBoth;
-  if (node.edge().roundabout()) {
-    return FormOfWay::ROUNDABOUT;
-  } else if (use == TripLeg::kRampUse || use == TripLeg::kTurnChannelUse) {
-    return FormOfWay::SLIPROAD;
-  } else if (rc == RoadClass::kMotorway && one_way) {
-    return FormOfWay::MOTORWAY;
-  } else if (rc <= RoadClass::kTertiary && one_way) {
-    return FormOfWay::MULTIPLE_CARRIAGEWAY;
-  } else if (rc <= RoadClass::kTertiary) {
-    return FormOfWay::SINGLE_CARRIAGEWAY;
-  } else {
-    return FormOfWay::OTHER;
-  }
-}
-
-std::vector<std::string> openlr_edges(const TripLeg& leg) {
-  const std::vector<PointLL>& points = decode<std::vector<PointLL>>(leg.shape());
-  std::vector<std::string> openlrs;
-  openlrs.reserve(leg.node_size());
-  for (const TripLeg::Node& node : leg.node()) {
-    // OpenLR for an edge is composed of two points with the following semantics:
-    // - FRC = relative importance of the road, so use the rank order of the FormOfWay enum.
-    // - Bearing of last point in the segment must point inward (bearing rotated by 180deg),
-    //   in-order to follow OpenLR conventions.
-    const FormOfWay fow = road_class_to_fow(node);
-    const unsigned char frc = static_cast<unsigned char>(fow);
-    unsigned char lfrcnp = frc;
-    const PointLL& start = points.at(node.edge().begin_shape_index());
```

Comment:
here we were making use of the edge without checking whether or not we have one. the last node in the leg will not have an edge as its the end of the route. so this was undefined behavior but also, semantically wrong. we should only have one openlr between every pair of nodes in the route. therefor the number of linear references in the output should be `leg.node_size() - 1`